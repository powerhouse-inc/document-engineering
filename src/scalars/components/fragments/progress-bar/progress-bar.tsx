import * as ProgressPrimitive from '@radix-ui/react-progress'
import React from 'react'
import { cn } from '../../../lib/utils.js'

const ProgressBar = React.forwardRef<
  React.ElementRef<typeof ProgressPrimitive.Root>,
  React.ComponentPropsWithoutRef<typeof ProgressPrimitive.Root> & {
    animationDuration?: number
  }
>(({ className, value, animationDuration = 500, ...props }, ref) => (
  <ProgressPrimitive.Root
    ref={ref}
    data-slot="progress-bar"
    className={cn('relative w-full h-3 overflow-hidden rounded bg-gray-100', className)}
    {...props}
  >
    <ProgressPrimitive.Indicator
      data-slot="progress-bar-indicator"
      className="w-full h-full bg-blue-900"
      style={{
        transform: `translateX(-${100 - (value ?? 0)}%)`,
        transition: `transform ${animationDuration}ms cubic-bezier(0.4, 0, 0.2, 1)`,
      }}
    />
  </ProgressPrimitive.Root>
))

ProgressBar.displayName = 'ProgressBar'

export { ProgressBar }
