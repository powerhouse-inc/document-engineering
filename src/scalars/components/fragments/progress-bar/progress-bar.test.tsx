import { describe, expect, it } from 'vitest'
import { render, screen } from '@testing-library/react'
import { ProgressBar } from './progress-bar.js'

describe('ProgressBar Component', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(<ProgressBar value={50} />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('should render with correct data-slot attributes', () => {
    render(<ProgressBar value={50} />)
    const progressBar = screen.getByRole('progressbar')
    const indicator = progressBar.querySelector('[data-slot="progress-bar-indicator"]')

    expect(progressBar).toHaveAttribute('data-slot', 'progress-bar')
    expect(indicator).toHaveAttribute('data-slot', 'progress-bar-indicator')
  })
})
