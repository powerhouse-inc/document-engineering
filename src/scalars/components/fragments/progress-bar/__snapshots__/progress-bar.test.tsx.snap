// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProgressBar Component > should match snapshot 1`] = `
<DocumentFragment>
  <div
    aria-valuemax="100"
    aria-valuemin="0"
    class="relative w-full h-3 overflow-hidden rounded-md bg-gray-100"
    data-max="100"
    data-slot="progress-bar"
    data-state="indeterminate"
    role="progressbar"
  >
    <div
      class="w-full h-full flex-1 transition-all bg-blue-900"
      data-max="100"
      data-slot="progress-bar-indicator"
      data-state="indeterminate"
      style="transform: translateX(-50%);"
    />
  </div>
</DocumentFragment>
`;
